packages/office-addin-telemetry/src/officeAddinTelemetry.ts

```diff
@@ -0,0 +1,251 @@
+import * as appInsights from "applicationinsights";
+import * as chalk from "chalk";
+import * as readLine from "readline-sync";
+import * as jsonData from "./telemetryJsonData";
+/**
+ * Specifies the telemetry infrastructure the user wishes to use
+ * @enum Application Insights: Microsoft Azure service used to collect and query through data
+ */
+export enum telemetryType {
+  applicationinsights = "applicationInsights",
+}
+/**
+ * Level controlling what type of telemetry is being sent
+ * @enum basic: basic level of telemetry, only sends errors
+ * @enum verbose: verbose level of telemetry, sends errors and events
+ */
+export enum telemetryLevel {
+  basic = "basic",
+  verbose = "verbose",
+}
+
+/**
+ * Telemetry object necessary for initialization of telemetry package
+ * @member groupName Telemetry Group name that will be written to the telemetry config file (i.e. telemetryJsonFilePath)
+ * @member projectName The name of the project that is using the telemetry package (e.g "generator-office")
+ * @member instrumentationKey Instrumentation key for telemetry resource
+ * @member promptQuestion Question displayed to user over opt-in for telemetry
+ * @member raisePrompt Specifies whether to raise telemetry prompt (this allows for using a custom prompt)
+ * @member telemetryLevel User's response to the prompt for telemetry and level of telemetry being sent
+ * @member telemetryType Telemetry infrastructure to send data
+ * @member testData Allows user to run program without sending actual data
+ */
+export interface ITelemetryOptions {
+  groupName: string;
+  projectName: string;
+  instrumentationKey: string;
+  promptQuestion: string;
+  raisePrompt: boolean;
+  telemetryLevel: telemetryLevel;
+  telemetryType: telemetryType;
+  testData: boolean;
+}
+
+/**
+ * Creates and initializes member variables while prompting user for telemetry collection when necessary
+ * @param telemetryObject
+ */
+export class OfficeAddinTelemetry {
+  private telemetryClient = appInsights.defaultClient;
+  private eventsSent: number = 0;
+  private exceptionsSent: number = 0;
+  private telemetryObject: ITelemetryOptions;
+
+  constructor(telemetryObj: ITelemetryOptions) {
```

Comment:
can we change telemetryObj to telemetryOptions to better match the type ITelemetryOptions?